Graphics/Implicit.hs

```diff
@@ -6,205 +6,212 @@
 
-module Graphics.Implicit (
-  -- * Types
-  W.ℝ,
-  W.ℝ2,
-  W.ℝ3,
-  SymbolicObj2 (),
-  SymbolicObj3 (),
-  W.ExtrudeMScale(C1, C2, Fn),
-
-  -- * Shared operations
-  P.Object (),
-  P.translate,
-  P.scale,
-  P.mirror,
-  P.complement,
-  P.union,
-  P.unionR,
-  P.intersect,
-  P.intersectR,
-  P.difference,
-  P.differenceR,
-  P.implicit,
-  P.shell,
-  P.outset,
-  P.emptySpace,
-  P.fullSpace,
-  P.withRounding,
-
-  -- * 2D primitive shapes
-  P.square,
-  P.rect,
-  P.circle,
-  P.polygon,
-
-  -- * 2D operations
-  P.rotate,
-  P.pack2,
-
-  -- * 3D primitive shapes
-  P.cube,
-  P.rect3,
-  P.sphere,
-  P.cylinder,
-  P.cylinder2,
-
-  -- * 3D operations
-  P.rotate3,
-  P.rotate3V,
-  P.pack3,
-
-  -- * Extrusions into 3D
-  P.extrude,
-  P.extrudeM,
-  P.extrudeOnEdgeOf,
-  P.rotateExtrude,
-
-  -- * OpenScad support
-  E.runOpenscad,
-
-  -- * 2D exporters
-  writeSVG,
-  writePNG2,
-  writeDXF2,
-  writeSCAD2,
-  writeGCodeHacklabLaser,
-
-  -- * 3D exporters
-  writeSTL,
-  writeBinSTL,
-  writeOBJ,
-  writeTHREEJS,
-  writeSCAD3,
-  writePNG3,
-
-  -- * Linear re-exports
-  L.V2(V2),
-  L.V3(V3)
-) where
-
-import Prelude(FilePath, IO)
+module Graphics.Implicit
+  ( -- * Types
+    W.ℝ,
+    W.ℝ2,
+    W.ℝ3,
+    SymbolicObj2 (),
+    SymbolicObj3 (),
+    W.ExtrudeMScale (C1, C2, Fn),
+
+    -- * Shared operations
+    P.Object (),
+    P.translate,
+    P.scale,
+    P.mirror,
+    P.complement,
+    P.union,
+    P.unionR,
+    P.intersect,
+    P.intersectR,
+    P.difference,
+    P.differenceR,
+    P.implicit,
+    P.shell,
+    P.outset,
+    P.emptySpace,
+    P.fullSpace,
+    P.withRounding,
+
+    -- * 2D primitive shapes
+    P.square,
+    P.rect,
+    P.circle,
+    P.polygon,
+
+    -- * 2D operations
+    P.rotate,
+    P.pack2,
+
+    -- * 3D primitive shapes
+    P.cube,
+    P.rect3,
+    P.sphere,
+    P.cylinder,
+    P.cylinder2,
+
+    -- * 3D operations
+    P.rotate3,
+    P.rotate3V,
+    P.pack3,
+
+    -- * Extrusions into 3D
+    P.extrude,
+    P.extrudeM,
+    P.extrudeOnEdgeOf,
+    P.rotateExtrude,
+
+    -- * OpenScad support
+    E.runOpenscad,
+
+    -- * 2D exporters
+    writeSVG,
+    writePNG2,
+    writeDXF2,
+    writeSCAD2,
+    writeGCodeHacklabLaser,
+
+    -- * 3D exporters
+    writeSTL,
+    writeBinSTL,
+    writeOBJ,
+    writeTHREEJS,
+    writeSCAD3,
+    writePNG3,
+
+    -- * Linear re-exports
+    L.V2 (V2),
+    L.V3 (V3),
+  )
+where
 
 -- The primitive objects, and functions for manipulating them.
 -- MAYBEFIXME: impliment slice operation, regularPolygon and zsurface primitives.
-import Graphics.Implicit.Primitives as P (withRounding, rect, rect3, translate, scale, mirror, complement, union, intersect, difference, unionR, intersectR, differenceR, shell, extrude, extrudeM, extrudeOnEdgeOf, sphere, cube, circle, cylinder, cylinder2, square, polygon, rotateExtrude, rotate3, rotate3V, pack3, rotate, pack2, implicit, fullSpace, emptySpace, outset, Object)
```

Comment:
Is there a way for us to tell ormolu that a comment is attached to an import?

Reply:
Ah, I see I haven't fixed it yet.

You can put comments between import lines, and they will stick there (see 83461ee).  The crazy thing just happened because ormolu had to move a few imports from further down the list to the top, and couldn't decide whether to put the underneath the comments or on top.

Reply:
> Ah, I see I haven't fixed it yet.

Ah again: I *have* fixed the other place, this is a new one.

Before we're done, I should probably go through all the changes and see what else I find...  Can't trust machines!  :)